docs/articles/features/vstest.md

```diff
@@ -0,0 +1,75 @@
+
+## Caveats and things to know
+- The VSTest adapter will not call your application's entry point.
+  - If you use the entry point to customize how your benchmarks are run, you will need to do this through other means such as an assembly-level `IConfigSource`.
+  - For more about this, please read: [Setting a default configuration](#setting-a-default-configuration).
+- The benchmark measurements may be affected by the VSTest host and your IDE 
+  - If you want to have more accurate performance results, it is recommended to run benchmarks through the CLI instead without other processes on the machine impacting performance.
+  - This does not mean that the measurements are useless though, it will still be able to provide useful measurements during development when comparing different approaches.
+- The test adapter will not display or execute benchmarks if optimizations are disabled.
+  - Please ensure you are compiling in Release mode or with `Optimize` set to true.
+  - Using an `InProcess` toolchain will let you run your benchmarks with optimizations disabled and will let you attach the debugger as well.
+- The test adapter will generate an entry point for you automatically
+  - The generated entry point will pass the command line arguments and the current assembly into `BenchmarkSwitcher`, so you can still use it in your CLI as well as in VSTest.
+  - This means you can delete your entry point and only need to define your benchmarks.
+  - If you want to use a custom entry point, you can still do so by setting `GenerateProgramFile` to `false` in your project file.
+
+## How to use it
+
+You need to install two packages into your benchmark project:
+
+- `BenchmarkDotNet.TestAdapter`: Implements the VSTest protocol for BenchmarkDotNet
+- `Microsoft.NET.Test.Sdk`: Includes all the pieces needed for the VSTest host to run and load the VSTest adapter.
+
+As mentioned in the caveats section, `BenchmarkDotNet.TestAdapter` will generate an entry point for you automatically, so if you have an entry point already you will either need to delete it or set `GenerateProgramFile` to `false` in your project file to continue using your existing one.
+
+After doing this, you can set your build configuration to `Release`, run a build, and you should be able to see the benchmarks in your IDE's VSTest integration.
+
+## Setting a default configuration
+
+Previously, it was common for the default configuration to be defined inside the entry point. Since the entry point is not used when running benchmarks through VSTest, the default configuration must be specified using a custom `IConfigSource` attribute instead that is set on the assembly. 
+
+First, create a custom attribute that implements `IConfigSource` like below, making sure that it has `Assembly` as one of the attribute targets:
+
+```csharp
+class MyDefaultConfigSourceAttribute : Attribute, IConfigSource
```

Comment:
We already have a type for that:

https://github.com/dotnet/BenchmarkDotNet/blob/b3b2d913a9de86147167f3fe65d770969c9541a5/src/BenchmarkDotNet/Attributes/ConfigAttribute.cs#L6-L11

Both solutions require to create new dedicated types. But IMO creating a type that just derives from `ManualConfig` is easier. Example:

https://github.com/dotnet/BenchmarkDotNet/blob/b3b2d913a9de86147167f3fe65d770969c9541a5/tests/BenchmarkDotNet.IntegrationTests.EnabledOptimizations/OptimizationsEnabledInCsproj.cs#L8-L18

Reply:
Thanks! just tested it out and that all works fine with the VSTest adapter